packages/flutter/lib/src/material/circle_avatar.dart

```diff
@@ -179,7 +179,14 @@ class CircleAvatar extends StatelessWidget {
       duration: kThemeChangeDuration,
       decoration: new BoxDecoration(
         color: effectiveBackgroundColor,
-        image: backgroundImage != null ? new DecorationImage(image: backgroundImage) : null,
+        image: backgroundImage != null
+			? new DecorationImage(
+				image: backgroundImage,
+				// Otherwise rectangular images are shrunk down
+				// and do not cover the entire avatar.
+				fit: BoxFit.cover
+			),
+			: null,
```

Comment:
Let's indent this slightly differently so it's more consistent with the rest of the code, e.g.:

```dart
        image: backgroundImage == null ? null :
          new DecorationImage(
            image: backgroundImage,
            fit: BoxFit.cover,
          ),
```